nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractDatabaseFetchProcessor.java

```diff
@@ -413,7 +400,7 @@ protected static String getLiteralByType(int type, String value, String database
             case TIMESTAMP:
                 // Timestamp literals in Oracle need to be cast with TO_DATE
                 if ("Oracle".equals(databaseType)) {
-                    return "to_date('" + value + "', 'yyyy-mm-dd HH24:MI:SS')";
+                    return "timestamp '" + value + "'";
```

Comment:
This probably breaks backwards compatibility if the processor has already stored a Date (no time-of-day) into the StateMap. To preserve backwards compatibility, we should check if the value is just a Date and use the "date" coercion instead of the "timestamp" coercion. If that succeeds, then the next maximum value observed should be stored as a full timestamp and the "date" coercion should no longer be executed